### PR TITLE
Embrace ESM/CJS standards a bit better

### PR DESCRIPTION
**Pull request by SanderElias**
# small improvement to bundling

- updated build to provide more clear naming for output files
- Remove unneeded ESM  package,
- Add esbuild package
- Use mjs/cjs extensions for the `patch-global.*` 
- add tests to see if cjs/esm loads an works.

### CODE REVIEW

File: test/can-load-as-cms.cjs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+require( "../dist/index.cjs");
+const test  = require( "ava"); 
+
+const baseURL = 'https://example.com';
+
```
Comment (kenchris) [inline test/can-load-as-cms.cjs:6]: Could we remove the extra newline?

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6137 @@
+{
+  "name": "urlpattern-polyfill",
```
Comment (kenchris) [inline package-lock.json]: I don't think we want to checkin the package-lock

**Review by kenchris — APPROVED** [review]
approved if we remove the package-lock :)

**Review by kenchris — APPROVED** [review]

**Comment by SanderElias** [comment]
closng in favor of #76

**Pull request closed without merging**